# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/svg/SVGPatternFrame.cpp
# Commit: b16351df250c
# Full Hash: b16351df250c572a14efa126f7702fe104ac9fd7
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-07-22 21:26:48
# Regressor Bug: 1843945
# File Overlap Count: 1
# Description:
#   Bug 1843945 - optimise SVGObserverUtils r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D183802
# ==============================================================================

diff -r 7e8063ebfd83 -r b16351df250c layout/svg/SVGPatternFrame.cpp
--- a/layout/svg/SVGPatternFrame.cpp	Sat Jul 22 10:03:30 2023 +0000
+++ b/layout/svg/SVGPatternFrame.cpp	Sat Jul 22 14:06:38 2023 +0000
@@ -65,7 +65,7 @@
        aAttribute == nsGkAtoms::width || aAttribute == nsGkAtoms::height ||
        aAttribute == nsGkAtoms::preserveAspectRatio ||
        aAttribute == nsGkAtoms::viewBox)) {
-    SVGObserverUtils::InvalidateDirectRenderingObservers(this);
+    SVGObserverUtils::InvalidateRenderingObservers(this);
   }
 
   if ((aNameSpaceID == kNameSpaceID_XLink ||
@@ -75,7 +75,7 @@
     SVGObserverUtils::RemoveTemplateObserver(this);
     mNoHRefURI = false;
     // And update whoever references us
-    SVGObserverUtils::InvalidateDirectRenderingObservers(this);
+    SVGObserverUtils::InvalidateRenderingObservers(this);
   }
 
   return SVGPaintServerFrame::AttributeChanged(aNameSpaceID, aAttribute,